src/main/scala/midas/widgets/Print.scala

```diff
@@ -55,102 +54,48 @@ class PrintRecordEndpoint extends Endpoint {
-    }
-    wires.ready || !valid
-  }
-  fire := (prints foldLeft (io.hPort.toHost.hValid && io.tReset.valid))(_ && _)
-  io.tReset.ready := fire
-  io.hPort.toHost.hReady := fire
-  // We don't generate tokens
-  io.hPort.fromHost.hValid := true.B
-  when (fire) {
-    cycles := Mux(io.tReset.bits, 0.U, cycles + 1.U)
-  }
-
-  override def genHeader(base: BigInt, sb: StringBuilder) {
-    import CppGenerationUtils._
-    sb.append(genComment("Print Widget"))
-    sb.append(genMacro("PRINTS_NUM", UInt32(printPort.elements.size)))
-    sb.append(genMacro("PRINTS_CHUNKS", UInt32(prints.size)))
-    sb.append(genMacro("PRINTS_ENABLE", UInt32(base + enableAddr)))
-    sb.append(genMacro("PRINTS_COUNT_ADDR", UInt32(base + countAddrs.head)))
-    sb.append(genArray("PRINTS_WIDTHS", printPort.elements.toSeq map (x => UInt32(x._2.getWidth))))
-    if (!p(HasDMAChannel)) {
-      sb.append(genArray("PRINTS_DATA_ADDRS", printAddrs.toSeq map (x => UInt32(base + x))))
-    } else {
-      sb.append(genMacro("HAS_DMA_CHANNEL"))
-    }
-  }
-
-  genCRFile()
+  //val printPort = io.hPort.hBits
+  //val fire = Wire(Bool())
+  //val cycles = Reg(UInt(48.W))
+  //val enable = RegInit(false.B)
+  //val enableAddr = attach(enable, "enable")
+  //val printAddrs = collection.mutable.ArrayBuffer[Int]()
+  //val countAddrs = collection.mutable.ArrayBuffer[Int]()
+  //val channelWidth = if (p(HasDMAChannel)) io.dma.get.nastiXDataBits else io.ctrl.nastiXDataBits
+  //val printWidth = (printPort.elements foldLeft 56)(_ + _._2.getWidth - 1)
+  //val valid = (printPort.elements foldLeft false.B)( _ || _._2(0))
+  //val ps = printPort.elements.toSeq map (_._2 >> 1)
+  //val vs = printPort.elements.toSeq map (_._2(0))
+  //val data = Cat(Cat(ps.reverse), Cat(vs.reverse) | 0.U(8.W), cycles)
+  ///* FIXME */ if (p(HasDMAChannel)) assert(printWidth <= channelWidth)
+
+  //fire := (prints foldLeft (io.hPort.toHost.hValid && io.tReset.valid))(_ && _)
+  //io.tReset.ready := fire
+  //io.hPort.toHost.hReady := fire
+  //// We don't generate tokens
+  //io.hPort.fromHost.hValid := true.B
+  //when (fire) {
+  //  cycles := Mux(io.tReset.bits, 0.U, cycles + 1.U)
+  //}
+
+  //override def genHeader(base: BigInt, sb: StringBuilder) {
+  //  import CppGenerationUtils._
+  //  sb.append(genComment("Print Widget"))
+  //  sb.append(genMacro("PRINTS_NUM", UInt32(printPort.elements.size)))
+  //  sb.append(genMacro("PRINTS_CHUNKS", UInt32(prints.size)))
+  //  sb.append(genMacro("PRINTS_ENABLE", UInt32(base + enableAddr)))
+  //  sb.append(genMacro("PRINTS_COUNT_ADDR", UInt32(base + countAddrs.head)))
+  //  sb.append(genArray("PRINTS_WIDTHS", printPort.elements.toSeq map (x => UInt32(x._2.getWidth))))
+  //  if (!p(HasDMAChannel)) {
+  //    sb.append(genArray("PRINTS_DATA_ADDRS", printAddrs.toSeq map (x => UInt32(base + x))))
+  //  } else {
+  //    sb.append(genMacro("HAS_DMA_CHANNEL"))
+  //  }
+  //}
+
+  //genCRFile()
```

Comment:
Why is this all commented out now?

Reply:
Print synthesis is not functional yet, so it was easier to comment out the code for now.